src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -214,6 +214,7 @@ protected override IEnumerable<ITypeSymbol> InferTypesWorker_DoNotCallDirectly(i
                     (IfStatementSyntax ifStatement) => InferTypeInIfStatement(ifStatement, token),
                     (InitializerExpressionSyntax initializerExpression) => InferTypeInInitializerExpression(initializerExpression, previousToken: token),
                     (LockStatementSyntax lockStatement) => InferTypeInLockStatement(lockStatement, token),
+                    (MemberAccessExpressionSyntax memberAccessExpression) => InferTypeInMemberAccessExpression(memberAccessExpression),
```

Comment:
this should take a token.   If there's a token, then you can do that work that causes this to think we should be 'bool' in that catch/when context.  

(though i'm not sure if this is what's causing us to infer 'bool' in the when context tbh.